ggml-metal.m

```diff
@@ -2237,31 +2226,24 @@ static bool ggml_metal_graph_compute(
 #endif
             }
 
-            if (encoder != nil) {
-                [encoder endEncoding];
-                encoder = nil;
-            }
+            [encoder endEncoding];
 
             [command_buffer commit];
+
+            [command_buffer waitUntilCompleted];
```

Comment:
The only concern I have is that we used to block one thread at max, now we block up to n_cb threads, this can lead to thread exhaustion in theory.

Reply:
That's a good point, it's probably worth keeping the second "status check" loop around. That would also remove any worries about the theoretical race issue in `all_buffers_succeeded`. I'll tweak this now 👍